app/src/main/java/com/boolder/boolder/data/database/repository/TickRepository.kt

```diff
@@ -0,0 +1,75 @@
+package com.boolder.boolder.data.database.repository
+
+import com.boolder.boolder.data.database.dao.AreaDao
+import com.boolder.boolder.data.database.dao.ProblemDao
+import com.boolder.boolder.data.database.dao.TickDao
+import com.boolder.boolder.data.database.entity.AreaWithProblems
+import com.boolder.boolder.data.database.entity.ProblemWithAreaName
+import com.boolder.boolder.data.database.entity.TickEntity
+
+class TickRepository(
+    private val tickDao: TickDao,
+    private val problemDao: ProblemDao,
+    private val areaDao: AreaDao,
+) {
+
+    suspend fun insertTick(tick: TickEntity) {
+        tickDao.insertTick(tick)
+    }
+
+    suspend fun loadById(tickId: Int): TickEntity? {
+        return tickDao.loadById(tickId)
+    }
+
+    suspend fun deleteById(tickId: Int){
+        return tickDao.deleteById(tickId)
+    }
+
+    suspend fun deleteAll(){
+        return tickDao.deleteAll()
+    }
+
+    suspend fun getProblemsPerArea(): List<AreaWithProblems> {
+        val tickEntities = tickDao.getAllIds()
+        val problemIds = tickEntities.map { it.id }
+        val problems = problemDao.loadAllByIds(problemIds)
+        val areaWithProblemsList = mutableListOf<AreaWithProblems>()
+
+        val areaIds = problems.map { it.areaId }
+        val areas = areaDao.loadAllByIds(areaIds.distinct())
+
+        // Group problems by area ID
+        val problemsByAreaId = problems.groupBy { it.areaId }
```

Comment:
Seems like this work should be delegated to a SQLite request instead of Kotlin code. It would be more efficient in terms of processing time. But the answer is not that simple, it raises the question about having a common database for the static data and the user's data or not (@nmondollot tell me if I am wrong, but the problems are updated by updating the whole database at once right?).

Reply:
I am also not sure about this one. I was thinking about added a "tick" flag to each entry in the "static database" thus essentially merging the databases. Although it makes the SQL queries a lot clearer I do think it could be tricky to update the static part when a new problem/area gets added.

Reply:
> (@nmondollot tell me if I am wrong, but the problems are updated by updating the whole database at once right?).

Yes, I update the database by replacing the boolder.db file with a brand new file.

> But the answer is not that simple, it raises the question about having a common database for the static data and the user's data or not

On iOS I use a separate database for ticks/favorites.
It works and it's simple. But we may run into performance issues in the future (?)

Reply:
I just pushed some changes where I added a state to each logged problem. Such that it can either be a favourite or a tick. Do you separate ticks/favourites into two databases?  If so can a problem be in both?